src/main/sd/init.c

```diff
@@ -0,0 +1,310 @@
+#include "sd.h"
+#include "debug.h"
+#include "pin.h"
+#include "spi.h"
+
+static bool sd_card_initialized = false;
+uint8_t csd_register[16];
+static uint8_t sd_state;
+mext2_command* command;
+
+#define SD_NOT_DETERMINED   0
+#define SD_V1X              1
+#define SD_V2X              2
+#define SD_V2XHCXC          3
+
+uint8_t sd_version = SD_NOT_DETERMINED;
+
+
+#define COMMAND_SOFTWARE_RESET      (uint8_t)0x00    //CMD0
+#define COMMAND_CHECK_VOLTAGE       (uint8_t)0x08    //CMD8
+#define COMMAND_READ_OCR            (uint8_t)0x3A    //CMD58
+#define COMMAND_BEFORE_INIT_PROCESS (uint8_t)0x37    //CMD55
+#define COMMAND_INIT_PROCESS        (uint8_t)0x29    //ACMD41
+#define COMMAND_READ_CSD            (uint8_t)0x09    //CMD09
+
+#define SD_INIT_OK              (uint8_t)0x00
+#define SD_INIT_NOK             (uint8_t)0x01
+#define SD_IDLE                 (uint8_t)0x02
+#define SD_ERROR                (uint8_t)0x03
+#define SD_INITIALIZED          (uint8_t)0x04
+
+#define SD_CONFIGURE_PINS       (uint8_t)0x05
+#define SD_RESET_SOFTWARE       (uint8_t)0x06
+#define SD_CHECK_VOLTAGE        (uint8_t)0x07
+#define SD_READ_OCR             (uint8_t)0x08
+#define SD_PREPARE_INIT_PROCESS (uint8_t)0x09
+#define SD_START_INIT_PROCESS   (uint8_t)0x0a
+#define SD_READ_OCR_AGAIN       (uint8_t)0x0b
+#define SD_READ_CSD             (uint8_t)0x0c
+
+bool check_sd_version = false;
+uint8_t csd_register[16];
```

Comment:
More globals to discuss

Reply:
csd_register moved to read_CSD_register function